### ISSUE TITLE
Home Assistant addon

### ISSUE THREAD

**Issue by postlund** [TradeFace/tuyagateway#77]
Just got a couple of tuya devices that I want to integrate into Home Assistant (locally). It would be really neat to have this gateway and GismoCaster installable as an add on, have you looked into supporting that? I'm a bit rusty regarding addon development, otherwise I would look into doing it myself.

**Comment by TradeFace** [comment]
Good idea. But I've no experience in creating addons for HA. (And frankly a bit too busy at the moment to look into that)

**Comment by sjthespian** [comment]
I'm planning on taking a look at creating an add-on once I get my environment working. The add-on part is fairly easy, I've done a few before (that never got published as they likely aren't useful to anyone but me).

**Comment by groft389** [comment]
> I'm planning on taking a look at creating an add-on once I get my environment working. The add-on part is fairly easy, I've done a few before (that never got published as they likely aren't useful to anyone but me).

Do you integrate GismoCaster into the same addon as TuyaGateway or in its own ? (And cool that you do this :D)

**Comment by sjthespian** [comment]
> Do you integrate GismoCaster into the same addon as TuyaGateway or in its own ? (And cool that you do this :D)

The easiest way to do it is as a separate add-on. Running multiple processes in a single container gets complicated, you have to run something like supervisord to manage them. I need to look at the source for some of the contributed add-ons to see if anyone has done that yet in the HA space.

**Comment by sjthespian** [comment]
I may have spoken too soon, I have found a couple of examples of working addons with multiple processes running in them. I have tuya-gateway running, so once I push that to git, I'll take a stab at adding GismoCaster into the add-on.

Version 0.1 should be up at https://github.com/sjthespian/addon-tuyagateway in just a few minutes.

**Comment by groft389** [comment]
> I may have spoken too soon, I have found a couple of examples of working addons with multiple processes running in them. I have tuya-gateway running, so once I push that to git, I'll take a stab at adding GismoCaster into the add-on.
> 
> Version 0.1 should be up at https://github.com/sjthespian/addon-tuyagateway in just a few minutes.

Good Job :D
Trying it out tomorrow.
If you manage to adding GismoCaster in the Same addon. Please do me a favour to be able to Change the Port and the IP of GismoCaster (Had to Change it from 127.0.0.1 to the Machines local IP 192.168.178.** otherwise IT wasnt possible for me to connect).

**Comment by sjthespian** [comment]
Version 0.2 is up and it has GizmoCaster included. I haven't done anything with it yet (since i'm still not exactly sure how to use it with my Tuya bulbs), but it's running and you can just use the `OPEN WEB UI` link in the add-on to connect to it. It is currently running on port 8111.

A couple of notes:
1. It takes a bit of time to start up GizomoCaster the first time as it has to create the database. It is in the /data partition though, so it should persist through restarts and reboots.
2. If you are running mosquitto for your MQTT broker on HA, you can use `core-mosquitto` for the mqtt_host setting in GizmoCaster.
3. It is possible to import an existing database, but the only way I know to do it now requires the `SSH & Web Terminal
` add-on (not the standard ssh add-on) and may need it to be running with protection mode disabled. You need to be able to copy your database into the running addon_local_tuyagateway container. It would

**Comment by sjthespian** [comment]
I just pushed out v0.3, it will update the mqtt settings in the GismoCaster database to match the settings in Tuya Gateway (i.e. the installed mqtt server or the settings in the add-on).  

On that note, i'm calling it a night.

**Comment by postlund** [comment]
This is great, I will try it out as soon as I have some free time. I would strongly suggest that you look at some of the addons that frenck has done, that bundles nginx. It lets you add support for ingress quite simply and also has a service concept (so you can run tuyagateway and gizmocast as services).

**Comment by sjthespian** [comment]
> This is great, I will try it out as soon as I have some free time. I would strongly suggest that you look at some of the addons that frenck has done, that bundles nginx. It lets you add support for ingress quite simply and also has a service concept (so you can run tuyagateway and gizmocast as services).

I had looked at those as I was working on this one, but I need to dig a bit deeper into how those services are configured and fired off. The HA add-on documentation really doesn't mention it.

I'm not sure I need to do anything to enable ingress at this point other than changing that setting in the config. There shouldn't be anything that requires configuration at this point.
